### PR TITLE
[FeedlyUser] factorize lazy streams and add simple examples

### PR DESCRIPTION
**Pull request by Mathieu4141**
- Factorizing the behaviour to lazily get streams in the `FeedlyUser` class. I wanted to add support to get the stream by names but with the previous caching system it needed to be done in 4 places
 - Adding 2 simple examples:
   - Streaming entries
   - Listing categories and tags

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -9,19 +9,19 @@
 import sys
```
Comment (Mathieu4141) [inline setup.py:9]: forgot to commit the black reformatting in previous PR

File: feedly/api_client/session.py [hunk 2]
```diff
@@ -125,25 +125,57 @@ def __exit__(self, exc_type, exc_val, exc_tb):
...
+
+    def make_api_request(
+        self,
+        relative_url: str,
+        method: str = None,
+        params: Dict[str, Any] = None,
+        data: Dict = None,
+        timeout: int = None,
+        max_tries: int = None,
+    ) -> Response:
```
Comment (Mathieu4141) [inline feedly/api_client/session.py:166]: I need to access the response headers to get the continuation in the IoC use case

File: feedly/api_client/data.py [hunk 3]
```diff
@@ -162,13 +175,62 @@ class Entry(FeedlyData):
...
+    def id2stream(self) -> Dict[str, StreamableT]:
+        return {stream.stream_id.content_id: stream for stream in self.streams}
+
+    @cached_property
+    def name2stream(self) -> Dict[str, StreamableT]:
+        return {stream["label"]: stream for stream in self.streams}
+
+    def get(self, name_or_id: Union[str, StreamIdBase]) -> StreamableT:
+        if isinstance(name_or_id, StreamIdBase):
+            return self.make_stream_from_id(name_or_id.content_id)
```
Comment (LouisTrezzini) [inline feedly/api_client/data.py:209]: is this safe? should we check that the other parts are matching our expectation?

File: feedly/api_client/data.py [hunk 4]
```diff
@@ -162,13 +175,62 @@ class Entry(FeedlyData):
...
+        except KeyError:
+            return self.get_from_name(name_or_id)
+
+    def get_from_name(self, name: str) -> StreamableT:
+        try:
+            return self.name2stream[name]
+        except KeyError:
+            raise ValueError(f"Stream `{name}` not found. Available streams: {list(self.name2stream)}") from None
+
+    def get_from_id(self, id: str) -> StreamableT:
```
Comment (LouisTrezzini) [inline feedly/api_client/data.py:222]: should this also raise a user friendly error?

File: README.md [hunk 5]
```diff
@@ -26,14 +26,17 @@ or you can create a new one when needed. It's a bit more efficient to keep it ar
 do choose to create clients as needed, you should pass in the user's ID in the constructor, 
 otherwise you'll incur a `/v3/profile` request. 
 
+## Examples setup
+
+To run [the examples](feedly/examples), you need first need to create a file containing you token in [feedly/examples/auth/access.token](feedly/examples/auth/access.token). You can also put your refresh token in [feedly/examples/auth/refresh.token](feedly/examples/auth/refresh.token).
+
```
Comment (LouisTrezzini) [inline README.md:32]: could we instead prompt the user for their token, and save it in the config folder? eg `~/.config/feedly/credentials.json`

Comment (LouisTrezzini) [reply README.md:32]: in the future, we could even request a developer access token automatically (using a browser flow)

File: feedly/examples/list_streams.py [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+from feedly.examples.utils import AUTH_DIR
+
+if __name__ == "__main__":
+    """
+    You need to setup your auth directory as described in the README of the library.
+    Alternatively, you can remove the `FileAuthStore` usage and replace it by the token directly, but you'll need to do
+     it in every example .
+     
+    This example will display your personal categories and tags.
+    Additionally, if you are part of an team, it will also display the enterprise ones.
```
Comment (LouisTrezzini) [inline feedly/examples/list_streams.py:13]: a team

**Review by LouisTrezzini — APPROVED** [review]

File: feedly/api_client/session.py [hunk 7]
```diff
@@ -44,7 +46,12 @@ class FileAuthStore(Auth):
     a file based token storage scheme
     """
 
-    def __init__(self, token_dir: Path, client_id: str = "feedlydev", client_secret: str = "feedlydev"):
+    def __init__(
+        self,
+        token_dir: Path = Path.home() / ".config/feedly",
+        client_id: str = "feedlydev",
+        client_secret: str = "feedlydev",
```
[the remark this answers is not in the document]
Comment (Mathieu4141) [reply feedly/api_client/session.py:53]: It's used in `make_api_request` to refresh the token http://github.com/feedly/python-api-client/blob/ef13f589f1437d4f291bb18c6e1ae5a94c58d0fa/feedly/api_client/session.py#L274-L274

[the remark this answers is not in the document]
Comment (LouisTrezzini) [reply feedly/api_client/session.py:53]: sorry, I thought you added these lines
Interesting, something to consider in a follow up PR

**Pull request merged**